code/obj/nuclearreactor/reactorcontrol.dm

```diff
@@ -0,0 +1,193 @@
+			src.UpdateIcon()
+
+
+
+
+	ui_interact(mob/user, datum/tgui/ui)
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if (!ui)
+			ui = new(user, src, "TurbineControl", src.name)
+			ui.open()
+
+	ui_static_data(mob/user)
+		. = list(
+		)
+
+	ui_data(mob/user)
+		. = list(
+			"rpm" = turbine_handle?.RPM,
+			"load" = turbine_handle?.stator_load,
+			"power" = turbine_handle?.lastgen,
+			"volume" = turbine_handle?.flow_rate,
+			"history" = src.history,
+			"overspeed" = turbine_handle?.overspeed,
+			"overtemp" = turbine_handle?.overtemp,
+		)
+
+	ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+		. = ..()
+		if(. || QDELETED(src.turbine_handle))
+			src.turbine_handle = null
+			return
+
+		switch(action)
+			if("loadChange")
+				var/x = params["newVal"]
+				src.turbine_handle.stator_load = min(max(x,1),10e30)
+				logTheThing("station", src, null, "[src.turbine_handle] stator load configured to [x] by [ui.user]")
+			if("volChange")
+				var/x = params["newVal"]
+				src.turbine_handle.flow_rate = min(max(x,1),10e5)
+				logTheThing("station", src, null, "[src.turbine_handle] flow rate configured to [x] by [ui.user]")
+/////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+/obj/machinery/power/nuclear/reactor_control
+	name = "Reactor Control Computer"
+	desc = "A computer for configuring and monitoring the a nuclear reactor."
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "reactor_stats"
+	density = TRUE
+	anchored = TRUE
+	var/obj/machinery/atmospherics/binary/nuclear_reactor/reactor_handle = null
+
+	process()
+		. = ..()
+		if(QDELETED(reactor_handle))
+			reactor_handle = null
+			var/datum/powernet/powernet = src.get_direct_powernet()
+			if(!powernet) return
+			for(var/obj/machinery/power/terminal/netlink/N in powernet.nodes)
+				if(istype(N.master,/obj/machinery/atmospherics/binary/nuclear_reactor))
+					src.reactor_handle = N.master
+
+
+	ui_interact(mob/user, datum/tgui/ui)
+		if(QDELETED(src.reactor_handle))
+			boutput(user,"No connection to reactor!")
+			ui?.close()
+			return
+		ui = tgui_process.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "NuclearReactor")
+			ui.open()
+
+	ui_static_data(mob/user)
+		. = list(
+			"gridW" = length(src.reactor_handle.component_grid),
+			"gridH" = length(src.reactor_handle.component_grid[1]),
+			"emptySlotIcon" = icon2base64(icon('icons/misc/reactorcomponents.dmi',"empty"))
+		)
+
+	ui_data(mob/user)
+		var/can_see_neutrons = isadmin(user)
+		var/control_rod_level = 0
+		var/control_rod_level_true = 0
+		var/control_rod_count = 0
+		var/comps[length(reactor_handle.component_grid)][length(src.reactor_handle.component_grid[1])]
+		for(var/x=1 to length(src.reactor_handle.component_grid))
+			for(var/y=1 to length(src.reactor_handle.component_grid[1]))
+				if(src.reactor_handle.component_grid[x][y])
+					var/obj/item/reactor_component/comp = src.reactor_handle.component_grid[x][y]
```

Comment:
you've checked the performance of this, right?

Reply:
I have, it's not bad. It's only 49 ops. The real performance issue is the neutrons, which is why the reactor explodes when the radiation level goes over a threshold, since it indicates an exponential process of generating more neutrons.

Profiling wise though, the reactor is surprisingly low cost except when it's right about to explode, and that state only lasts a couple machine ticks.